packages/slice-machine/src/features/customTypes/customTypesConfig.ts

```diff
@@ -23,3 +25,22 @@ export const CUSTOM_TYPES_CONFIG = {
     Icon: DatabaseIcon,
   },
 };
+
+export function matchesBuilderPagePathname(
+  pathname: string,
+  customTypeId: string
+): boolean {
+  return Object.values(CUSTOM_TYPES_CONFIG).some(({ getBuilderPagePathname }) =>
+    pathname.startsWith(getBuilderPagePathname(customTypeId))
+  );
+}
+
+export function readBuilderPageDynamicSegment(
```

Comment:
I feel what you want is just a "find" no?

```suggestion
export function readBuilderPageDynamicSegment(
  query: NextRouter["query"]
): string | undefined {
  const customTypesConfigValues = Object.values(CUSTOM_TYPES_CONFIG);

  const matchingConfigValue = customTypesConfigValues.find(
    ({ builderPageDynamicSegment }) =>
      typeof query[builderPageDynamicSegment] === "string"
  );

  return matchingConfigValue ? query[matchingConfigValue.builderPageDynamicSegment] : undefined;
}
```

I always favor dedicated array utils function over simple "for" to explain exactly what you're doing to the other devs.